Mathlib/NumberTheory/EulerProduct/Basic.lean

```diff
@@ -0,0 +1,170 @@
+when `f` is completely multiplicative (implemented as `f : ℕ →*₀ F`).
+
+An intermediate step is `EulerProduct.summable_and_hasSum_smoothNumbers_prod_primesBelow_tsum`
+(and its variant `EulerProduct.summable_and_hasSum_smoothNumbers_prod_primesBelow_tsum_geometric`),
+which relates the finite product over primes to the sum of `f n` over `N`-smooth `n`.
+
+## Tags
+
+Euler product
+-/
+
+variable {F : Type*} [NormedField F] [CompleteSpace F]
+
+/-- If `f` is multiplicative and summable, then its values at natural numbers `> 1`
+have norm strictly less than `1`. -/
+lemma Summable.norm_lt_one {f : ℕ →* F} (hsum : Summable f) {p : ℕ} (hp : 1 < p) :
+    ‖f p‖ < 1 := by
+  refine summable_geometric_iff_norm_lt_1.mp ?_
+  simp_rw [← map_pow]
+  exact hsum.comp_injective <| Nat.pow_right_injective hp
+
+
+open scoped Topology
+
+open BigOperators
+
+-- We work with series indexed by the natural numbers
+-- and with terms in a complete normed field `F`.
+variable {F : Type*} [NormedField F] [CompleteSpace F] {f : ℕ → F}
+
+-- We assume that `f` is *multiplicative* in the sense of arithmetic functions,
+-- i.e., multiplicative on coprime elements.
+-- The condition `f 1 = 1` is then equivalent to `f 1 ≠ 0`, but more convenient to use.
+variable (hf₁ : f 1 = 1) (hmul : ∀ {m n}, Nat.Coprime m n → f (m * n) = f m * f n)
+
+lemma Nat.map_prime_pow_mul {p : ℕ} (hp : p.Prime) (e : ℕ) {m : p.smoothNumbers} :
+    f (p ^ e * m) = f (p ^ e) * f m :=
+  hmul <| Nat.Coprime.pow_left _ <| hp.smoothNumbers_coprime <| Subtype.mem m
+
+namespace EulerProduct
+
+open Nat in
+/-- We relate a finite product over primes to an infinite sum over smooth numbers. -/
+lemma summable_and_hasSum_smoothNumbers_prod_primesBelow_tsum
+    (hsum : ∀ {p : ℕ}, p.Prime → Summable (fun n : ℕ ↦ ‖f (p ^ n)‖)) (N : ℕ) :
+    Summable (fun m : N.smoothNumbers ↦ ‖f m‖) ∧
+      HasSum (fun m : N.smoothNumbers ↦ f m) (∏ p in N.primesBelow, ∑' (n : ℕ), f (p ^ n)) := by
+  induction' N with N ih
+  · rw [smoothNumbers_zero, primesBelow_zero, Finset.prod_empty]
+    exact ⟨Set.Finite.summable (Set.finite_singleton 1) (‖f ·‖), hf₁ ▸ hasSum_singleton 1 f⟩
+  · rw [primesBelow_succ]
+    split_ifs with hN
+    · constructor
+      · simp_rw [← (equivProdNatSmoothNumbers hN).summable_iff, Function.comp_def,
+                 equivProdNatSmoothNumbers_apply', map_prime_pow_mul hmul hN, norm_mul]
+        apply summable_mul_of_summable_norm (f := fun (x : ℕ) ↦ ‖f (N ^ x)‖)
+          (g := fun (x : N.smoothNumbers) ↦ ‖f x.val‖) <;>
+          simp_rw [norm_norm]
+        exacts [hsum hN, ih.1]
+      · simp_rw [Finset.prod_insert (not_mem_primesBelow N),
+                 ← (equivProdNatSmoothNumbers hN).hasSum_iff, Function.comp_def,
+                 equivProdNatSmoothNumbers_apply', map_prime_pow_mul hmul hN]
+        apply (hsum hN).of_norm.hasSum.mul ih.2
+        -- `exact summable_mul_of_summable_norm (hsum hN) ih.1` gives a time-out
```

Comment:
Oddly, it looks like 
```lean
have := summable_mul_of_summable_norm (hsum hN) ih.1
exact this
```
works fine. Maybe worth asking on zulip about this?

Reply:
See [here](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Proving.20a.20sum.20over.20smooth.20numbers.20is.20summable/near/400602151).

Reply:
Oh well; another one for the "known bugs" file.